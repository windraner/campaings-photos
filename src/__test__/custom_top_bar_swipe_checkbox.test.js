import React from 'react';
import CustomTopBarSwipeCheckbox from '../components/main_container/top_bar/custom_top_bar_swipe_checkbox/custom_top_bar_swipe_checkbox';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const mock = {
  isChecked: true,
  switchHandler() {},
};

describe('CustomTopBarSwipeCheckbox', () => {
  it('Check for value is setted', () => {
    const wrapper = shallow(<CustomTopBarSwipeCheckbox {...mock} />);
    expect(wrapper.find('.campaigns-photos__switch-input').props().checked).toBe(true);
    wrapper.setProps({ isChecked: false });
    expect( wrapper.find('.campaigns-photos__switch-input').props().checked).toBe(false);
  });
  it('Check for callback is called', () => {
    const onClickFunc = jest.fn();
    const wrapper = shallow(<CustomTopBarSwipeCheckbox {...mock} switchHandler={onClickFunc} />);
    const button = wrapper.find('.campaigns-photos__switch-input');
    button.simulate('change');
    expect(onClickFunc).toBeCalled();
    expect(onClickFunc.mock.calls).toEqual([['isFilterShown']]);
  });
  it('matches the snapshot', () => {
    const tree = shallow(<CustomTopBarSwipeCheckbox {...mock} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
