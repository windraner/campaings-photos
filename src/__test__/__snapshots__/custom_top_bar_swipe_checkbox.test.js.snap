// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTopBarSwipeCheckbox matches the snapshot 1`] = `
<label
  className="d-flex align-items-center m-0"
>
  <input
    checked={true}
    className="campaigns-photos__custom-checkbox mr-2"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="campaigns-photos__customTopBarCheckbox-text campaigns-photos__customTopBarCheckbox-text__active"
  >
    Select all
  </span>
</label>
`;
